src/runtime_src/core/pcie/emulation/cpu_em/generic_pcie_hal2/shim.cxx

```diff
@@ -1455,13 +1454,13 @@ namespace xclcpuemhal2
 
   /*********************************** Utility ******************************************/
 
-  xclemulation::drm_xocl_bo *CpuemShim::xclGetBoByHandle(unsigned int boHandle)
+  std::shared_ptr<xclemulation::drm_xocl_bo> CpuemShim::xclGetBoByHandle(unsigned int boHandle)
   {
     auto it = mXoclObjMap.find(boHandle);
     if (it == mXoclObjMap.end())
       return nullptr;
 
-    xclemulation::drm_xocl_bo *bo = (*it).second;
+    auto bo = (*it).second;
```

Comment:
```suggestion
    auto bo = it->second;
```

Reply:
Agreed. The code is updated accordingly.